### PR TITLE
fix: validate only one isbsvc implementation type is defined. Fixes #269

### LINKED ISSUE #269: Both redis and jetstream in isbsvc causes unexpected behavior
**Describe the bug**

Things can get into a weird state if you accidentally leave multiple isbsvc types in spec.

**To Reproduce**

1. The following causes only redis to run:
```yaml
apiVersion: numaflow.numaproj.io/v1alpha1
kind: InterStepBufferService
metadata:
  name: default
spec:
  jetstream:
    version: latest
  redis:
    native:
      version: 6.2.6
```
2. If you then apply the following, both redis and jetstream run:
```yaml
apiVersion: numaflow.numaproj.io/v1alpha1
kind: InterStepBufferService
metadata:
  name: default
spec:
  jetstream:
    version: latest
```

**Expected behavior**
Applied yaml should either error during isbsvc validation or be reflected in cluster.

**Environment (please complete the following information):**

- Kubernetes: 1.25.3
- Numaflow: head of main

**Additional context**

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We often sort issues this way to know what to prioritize.

### PR DESCRIPTION
**Pull request by dseapy**
Signed-off-by: David Seapy <dseapy@gmail.com>

Fixes #269

Validates only one isbsvc implementation type is defined

### CODE REVIEW

**Comment by dseapy** [comment]
I think the intention is for only one to be defined, but if that's not the case & both should be able to be defined together then I can work on a different PR to address the issue.

**Comment by whynowy** [comment]
We will have a validatingwebhook provided to prevent this from happening - to switch isbsvc.

**Comment by dseapy** [comment]
> We will have a validatingwebhook provided to prevent this from happening - to switch isbsvc.

ok, so you are saying that's a future effort to implement a validating webhook that handles that?  If so sounds like I should create a separate ticket to track that and this PR should be ok as-is?

**Comment by whynowy** [comment]
> > We will have a validatingwebhook provided to prevent this from happening - to switch isbsvc.
> 
> 
> 
> ok, so you are saying that's a future effort to implement a validating webhook that handles that?  If so sounds like I should create a separate ticket to track that and this PR should be ok as-is?

I think so, thanks!

**Review by vigith — APPROVED** [review]

File: pkg/reconciler/isbsvc/validate.go [hunk 1]
```diff
@@ -8,6 +8,9 @@ import (
 
 // ValidateInterStepBufferService accepts an isbs and performs validation against it
 func ValidateInterStepBufferService(isbs *dfv1.InterStepBufferService) error {
+	if isbs.Spec.Redis != nil && isbs.Spec.JetStream != nil {
```
[the remark this answers is not in the document]
Comment (dseapy) [reply pkg/reconciler/isbsvc/validate.go:11]: good call, did the others in the same block to keep consistent.  hopefully that's ok, if not let me know i can revert those and only change the lines i added.

**Pull request merged**